File: app/src/main/java/vinova/henry/com/hotfilm/feature/trailer/YouTubeActivity.java
package vinova.henry.com.hotfilm.feature.trailer;

import android.os.Bundle;

import com.google.android.youtube.player.YouTubeBaseActivity;
import com.google.android.youtube.player.YouTubeInitializationResult;
import com.google.android.youtube.player.YouTubePlayer;
import com.google.android.youtube.player.YouTubePlayerView;

import butterknife.BindView;
import butterknife.ButterKnife;
import vinova.henry.com.hotfilm.R;
import vinova.henry.com.hotfilm.models.Trailer;

public class YouTubeActivity extends YouTubeBaseActivity {

    private static final String YOUTUBE_API_KEY = "AIzaSyBPjiGer_AHJYZat5aXgW4VBz7dHdCGxXQ";
    Trailer trailer;

    @BindView(R.id.youtube_player)
    YouTubePlayerView youtubePlayer;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_you_tube);
        ButterKnife.bind(this);

        Bundle ex = getIntent().getExtras();
        if (ex != null) {
            trailer = (Trailer) ex.get("Trailer");
        }

        youtubePlayer.initialize(YOUTUBE_API_KEY, new YouTubePlayer.OnInitializedListener() {
            @Override
            public void onInitializationSuccess(YouTubePlayer.Provider provider, final YouTubePlayer youTubePlayer, boolean b) {
                youTubePlayer.loadVideo(trailer.getKey());
                youTubePlayer.setPlayerStyle(YouTubePlayer.PlayerStyle.DEFAULT);
                youTubePlayer.setFullscreen(true);
                youTubePlayer.setShowFullscreenButton(false);
                youTubePlayer.setPlayerStateChangeListener(new YouTubePlayer.PlayerStateChangeListener() {
                    @Override
                    public void onLoading() {

                    }

                    @Override
                    public void onLoaded(String s) {

                    }

                    @Override
                    public void onAdStarted() {

                    }

                    @Override
                    public void onVideoStarted() {

                    }

                    @Override
                    public void onVideoEnded() {
                        YouTubeActivity.super.onBackPressed();
                    }

                    @Override
                    public void onError(YouTubePlayer.ErrorReason errorReason) {

                    }
                });
                youTubePlayer.setPlaybackEventListener(new YouTubePlayer.PlaybackEventListener() {
                    @Override
                    public void onPlaying() {

                    }

                    @Override
                    public void onPaused() {

                    }

                    @Override
                    public void onStopped() {

                    }

                    @Override
                    public void onBuffering(boolean b) {

                    }

                    @Override
                    public void onSeekTo(int i) {

                    }
                });
            }

            @Override
            public void onInitializationFailure(YouTubePlayer.Provider provider, YouTubeInitializationResult youTubeInitializationResult) {

            }
        });
    }
}

File: app/src/main/java/vinova/henry/com/hotfilm/feature/trailer/YouTubeActivity.kt
package vinova.henry.com.hotfilm.feature.trailer

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.View
import android.widget.LinearLayout
import com.google.android.youtube.player.YouTubeBaseActivity
import com.google.android.youtube.player.YouTubeInitializationResult
import com.google.android.youtube.player.YouTubePlayer
import kotlinx.android.synthetic.main.activity_you_tube.*
import vinova.henry.com.hotfilm.R
import vinova.henry.com.hotfilm.YOUTUBE_API_KEY
import vinova.henry.com.hotfilm.models.trailer.Trailer

class YouTubeActivity : YouTubeBaseActivity(), YouTubePlayer.OnInitializedListener, YouTubePlayer.OnFullscreenListener {

    private var youTubePlayer: YouTubePlayer? = null
    private var videoKey: String? = null
    private var trailerAdapter: TrailerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_you_tube)

        trailerAdapter = TrailerAdapter(onTrailerClicked)
        rvTrailer.adapter = trailerAdapter
        rvTrailer.layoutManager = LinearLayoutManager(this, LinearLayout.VERTICAL, false)

        intent.let {
            val trailers = it.extras.get("trailers") as ArrayList<Trailer>
            trailerAdapter?.setTrailers(trailers = trailers)
            val pos = it.extras.get("position") as Int
            videoKey = trailers[pos].key
            tvTitle.text = trailers[pos].name
        }

        youTubePlayerView.initialize(YOUTUBE_API_KEY, this)
        imBack.setOnClickListener { super.onBackPressed() }
    }

    override fun onInitializationSuccess(p0: YouTubePlayer.Provider?, p1: YouTubePlayer?, p2: Boolean) {
        this.youTubePlayer = p1
        p1?.addFullscreenControlFlag(YouTubePlayer.FULLSCREEN_FLAG_CUSTOM_LAYOUT)
        youTubePlayer?.setOnFullscreenListener(this)
        youTubePlayer?.cueVideo(videoKey)
        youTubePlayer?.play()
    }

    override fun onInitializationFailure(p0: YouTubePlayer.Provider?, p1: YouTubeInitializationResult?) {

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        if (requestCode == 1) {
            youTubePlayerView.initialize(YOUTUBE_API_KEY, this)
        }
    }

    override fun onFullscreen(p0: Boolean) {
        if (p0) {
            clToolBar.visibility = View.GONE
        } else {
            clToolBar.visibility = View.VISIBLE
        }
    }

    override fun onConfigurationChanged(newConfig: Configuration?) {
        super.onConfigurationChanged(newConfig)
        if (newConfig?.orientation == Configuration.ORIENTATION_LANDSCAPE) {
            clToolBar.visibility = View.GONE
        } else if (newConfig?.orientation == Configuration.ORIENTATION_PORTRAIT){
            clToolBar.visibility = View.VISIBLE
        }
    }

    private var onTrailerClicked = fun(videoKey: String?, videoName: String?) {
        this.videoKey = videoKey
        tvTitle.text = videoName
        youTubePlayer?.cueVideo(videoKey)
    }
}